src/main/java/com/esri/arcgisruntime/toolkit/skins/FeatureTemplatePickerTilePaneSkin.java

```diff
@@ -0,0 +1,329 @@
+ * snippet shows a selector for applying a style to feature template groups only when the picker has a horizontal
+ * orientation:</p>
+ * <pre>
+ * .feature-template-picker:horizontal .feature-template-group {
+ *
+ * }
+ * </pre>
+ *
+ * @since 100.7.0
+ */
+public final class FeatureTemplatePickerTilePaneSkin extends SkinBase<FeatureTemplatePicker> {
+
+  private static final PseudoClass HORIZONTAL_PSEUDO_CLASS = PseudoClass.getPseudoClass("horizontal");
+
+  private Pane contentPane = new VBox();
+  private final ScrollPane scrollPane = new ScrollPane();
+  private final ToggleGroup toggleGroup = new ToggleGroup();
+  private boolean swatchesNeedUpdate = false;
+
+  public FeatureTemplatePickerTilePaneSkin(FeatureTemplatePicker control) {
+    super(control);
+    // put everything in a scroll pane to handle sizing within the skin
+    getChildren().add(scrollPane);
+
+    updateOrientation();
+
+    // use a shared toggle group for all feature template items to keep track of the current selected item
+    toggleGroup.selectedToggleProperty().addListener((observable, oldValue, newValue) ->
+      control.setSelectedFeatureTemplateItem(newValue == null ? null : (FeatureTemplateItem) newValue.getUserData())
+    );
+    control.selectedFeatureTemplateItemProperty().addListener((observable, oldValue, newValue) -> {
+      if (newValue == null) {
+        toggleGroup.selectToggle(null);
+      } else {
+        // when selected programmatically, find its toggle in the toggle group and select it in the toggle group
+        Toggle matchingToggle = toggleGroup.getToggles().stream()
+            .filter(t -> (t.getUserData()).equals(newValue))
+            .findFirst()
+            .orElse(null);
+        toggleGroup.selectToggle(matchingToggle);
+      }
+    });
+
+    // add a tile pane for each feature template group to the view
+    control.getFeatureTemplateGroups().stream()
+        .map(this::createVBoxForFeatureTemplateGroup)
+        .forEach(tilePane -> contentPane.getChildren().add(tilePane));
+
+    // add or remove tile panes if the feature template group collection changes
+    control.featureTemplateGroupsProperty().addListener((ListChangeListener<FeatureTemplateGroup>) c -> {
+      while (c.next()) {
+        if (c.wasAdded()) {
+          // create vboxes for the each feature template group
+          List<VBox> added = c.getAddedSubList().stream()
+              .map(this::createVBoxForFeatureTemplateGroup)
+              .collect(Collectors.toList());
+          // add the vboxes to the view
+          contentPane.getChildren().addAll(c.getFrom(), added);
+        } else if (c.wasRemoved()) {
+          // remove the vboxes based on the
+          contentPane.getChildren().remove(c.getFrom(), c.getFrom() + c.getRemovedSize());
+        }
+      }
+    });
+
+    control.orientationProperty().addListener((observable, oldValue, newValue) ->
```

Comment:
I think if you don't actually use `newValue` you can do these kinds of listeners as 
```control.orientationProperty().addListener(o -> updateOrientation());```